UOP1_Project/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using UnityEngine;
+
+public class AudioManager : MonoBehaviour
+{
+    private AudioSource audioSource = default;
+
+    private void Awake()
+    {
+        PreventMultipleAudioManagers();
+        audioSource = GetComponent<AudioSource>();
+    }
+
+    public void Play(AudioClip audioClip)
+    {
+        audioSource.clip = audioClip;
+        audioSource.Play();
+    }
+
+    public void Stop()
+    {
+        audioSource.Stop();
+    }
+
+    private void PreventMultipleAudioManagers()
+    {
+        UnityEngine.Object[] otherAudioManagers = FindObjectsOfType(GetType());
+        if (otherAudioManagers.Length > 1)
+        {
+            gameObject.SetActive(false);
```

Comment:
why do you need to setActive as false if you are going to destroy it anyway?

Reply:
Good question :) `Destroy` does not actually destroy the object immediately. Therefore, inactivating the object first is a bit safer.

Reply:
I think SetActive would act from the next frame anyway, so it's not "safer".
If you're not sure about wheter another object is going to execute code in another Update before this one gets destroyed, I think it's best to eliminate the ambiguity than try to patch it with things like this. Safer in the long term!

Reply:
Fair enough - I removed the `SetActive`!